packages/lexical-playground/src/plugins/TableHoverActionsPlugin/index.tsx

```diff
@@ -0,0 +1,243 @@
+            const colCount = (
+              (table as TableNode).getChildAtIndex(0) as TableRowNode
+            )?.getChildrenSize();
+
+            const rowIndex = $getTableRowIndexFromTableCellNode(maybeTableCell);
+            const colIndex =
+              $getTableColumnIndexFromTableCellNode(maybeTableCell);
+
+            if (rowIndex === rowCount - 1) {
+              hoveredRowNode = maybeTableCell;
+            } else if (colIndex === colCount - 1) {
+              hoveredColumnNode = maybeTableCell;
+            }
+          }
+        }
+      });
+
+      if (tableDOMElement) {
+        const {
+          width: tableElemWidth,
+          y: tableElemY,
+          x: tableElemX,
+          right: tableElemRight,
+          bottom: tableElemBottom,
+          height: tableElemHeight,
+        } = (tableDOMElement as HTMLTableElement).getBoundingClientRect();
+
+        const {y: editorElemY} = anchorElem.getBoundingClientRect();
+
+        if (hoveredRowNode) {
+          setShownRow(true);
+          setPosition({
+            height: BUTTON_WIDTH_PX,
+            left: tableElemX,
+            top: tableElemBottom - editorElemY + 5,
+            width: tableElemWidth,
+          });
+        } else if (hoveredColumnNode) {
+          setShownColumn(true);
+          setPosition({
+            height: tableElemHeight,
+            left: tableElemRight + 5,
+            top: tableElemY - editorElemY,
+            width: BUTTON_WIDTH_PX,
+          });
+        }
+      }
+    },
+    50,
+    250,
+  );
+
+  useEffect(() => {
+    if (!shouldListenMouseMove) {
+      return;
+    }
+
+    document.addEventListener('mousemove', debouncedOnMouseMove);
+
+    return () => {
+      setShownRow(false);
+      setShownColumn(false);
+      debouncedOnMouseMove.cancel();
+      document.removeEventListener('mousemove', debouncedOnMouseMove);
+    };
+  }, [shouldListenMouseMove, debouncedOnMouseMove]);
+
+  editor.registerMutationListener(TableNode, (mutations) => {
+    editor.getEditorState().read(() => {
+      for (const [key, type] of mutations) {
+        switch (type) {
+          case 'created':
+            codeSetRef.current.add(key);
+            setShouldListenMouseMove(codeSetRef.current.size > 0);
+            break;
+
+          case 'destroyed':
+            codeSetRef.current.delete(key);
+            setShouldListenMouseMove(codeSetRef.current.size > 0);
+            break;
+
+          default:
+            break;
+        }
+      }
+    });
+  });
+
+  const insertAction = (insertRow: boolean) => {
+    editor.update(() => {
+      if (tableDOMNodeRef.current) {
+        const maybeTableNode = $getNearestNodeFromDOMNode(
+          tableDOMNodeRef.current,
+        );
+        maybeTableNode?.selectEnd();
+        if (insertRow) {
+          $insertTableRow__EXPERIMENTAL();
+          setShownRow(false);
+        } else {
+          $insertTableColumn__EXPERIMENTAL();
+          setShownColumn(false);
+        }
```

Comment:
No strong opinions. Your version is easy to read
```suggestion
        $insertTableColumn__EXPERIMENTAL();
        setShownColumn(insertRow);
```

Reply:
This doesn't work since it's different methods it's setShownRow and setShownColumn, not setShownColumn at all times.